tests/bucket_manager.py

```diff
@@ -73,35 +97,24 @@ def boto3_bucket(
 
     name = bucket_name or f"testrun-bucket-{uuid()}"
     # We only set the endpoint url if we're testing against a non-aws host
-    if port != 80:
-        s3_client = boto3.client(
-            "s3",
-            endpoint_url=endpoint_url,
-        )
-    else:
-        s3_client = boto3.client(
-            "s3",
-        )
-
-    s3_client.create_bucket(Bucket=name)
-
     if port != 80:
         s3_resource = boto3.resource(
             "s3",
             endpoint_url=endpoint_url,
+            aws_access_key_id=access_key,
+            aws_secret_access_key=secret_key,
+            region_name="us-east-1",
         )
     else:
         s3_resource = boto3.resource(
             "s3",
+            aws_access_key_id=access_key,
+            aws_secret_access_key=secret_key,
+            region_name="us-east-1",
         )
 
     bucket = s3_resource.Bucket(name)
-
-    yield bucket
-
-    for key in bucket.objects.all():
-        key.delete()
-    bucket.delete()
+    return bucket
```

Comment:
We don't we clean up the bucket anymore?

Reply:
I split the methods up into creating a bucket reference and creating the bucket itself, which also includes cleaning up the bucket. See [here](https://github.com/data-engineering-collective/minimalkv/pull/76/files#diff-2d7b489f836852c4047e7cbfb7b5f8cca1fd388d7acf20b86a965fb1feb9e299R72).